### ISSUE TITLE
Ignore flag for files and folders

### ISSUE THREAD

**Issue by pikeas** [microsoft/pylance-release#422]
It's very common to have a virtualenv folder in a project directory. Python files in a virtualenv should be skipped by Pylance, either by default (`.venv`, `site-packages`, etc) or via a configurable ignore/exclude setting.

**Comment by jakebailey** [comment]
We already ignore these files for purposes of workspace symbols, and the rest of the analysis is lazy and will only load files as needed to analyze the code you've written. Workspaces that contain their virtual environments are the main thing that we test with ourselves as that's the most common setup.

Can you describe what it is that isn't working about Pylance for you? Are you finding something to be slow? Are results showing up that shouldn't be there? Are diagnostics appearing in the wrong files?

**Comment by pikeas** [comment]
Thanks for the quick reply, @jakebailey! 

I have VS Code 1.49.2 (latest release) open into a folder containing a `.venv` folder, which is set as my active Python environment. Here's what I see in the Problems pane when I open, for example, `.venv/lib/python3.8/site-packages/pytest/__init__.py`:

<img width="707" alt="Screen Shot 2020-09-25 at 15 47 29" src="https://user-images.githubusercontent.com/686573/94322110-6d041880-ff46-11ea-9eb8-b75eab0bf7da.png">

Pylance appears to be analyzing this file. This happens for any `site-packages/**/*.py` file. It's rather distracting. If this is unexpected behavior, what setting(s) should I check? I don't see anything that seems relevant in `settings.json`.

**Comment by jakebailey** [comment]
Is your diagnostic mode set to "openFilesOnly"? That should be the default, though even in that mode I wouldn't expect it to behave this way.

**Comment by pikeas** [comment]
> Is your diagnostic mode set to "openFilesOnly"? That should be the default, though even in that mode I wouldn't expect it to behave this way.

Yes, I have: `"python.analysis.diagnosticMode": "openFilesOnly"`.

**Comment by jakebailey** [comment]
Did you open that file, or does the diagnostic appear if you've only opened one of the files in your project that imports it?

**Comment by pikeas** [comment]
I did some dig and found a clue. If I import a 3rd party module and then jump to a definition with cmd+click, Pylance does what I would expect. It seems to only trigger unexpectedly when I directly open a file via the file explorer. 

Open via file explorer: ❌ Pylance analyzes it
Add as import in project, close and reopen via explorer: ✅ Pylance ignores it
Add as import in project, open via cmd-click jump: ✅ Pylance ignores it
Remove import, close and reopoen: ❌ Pylance analyzes it

**Comment by benblo** [comment]
I too would like the option to exclude some directories.

I have a site-packages embedded in a subdir of my workspace directory (not a venv because reasons), set as excluded in the workspace settings, but added in `python.analysis.extraPaths` for autocompletion.
Today I tried enabling `"python.analysis.diagnosticMode": "workspace"` and was instantly flooded by errors from packages, making it extremely cumbersome to find reports from my own codebase.

The "hide excluded files" filter has no effect:
![image](https://user-images.githubusercontent.com/574187/98291963-3a6d2700-1fac-11eb-9f3c-f4831f7e0195.png)

I tried adding a `.vscode/settings.json` file to set `"python.analysis.diagnosticMode": "openFilesOnly"` only for that directory, to no avail... that would have been nice and generic.
Alternatively, it would be nice to have a way to declare something `"python.analysis.diagnosticMode.exclude":  = ["directory", "list"]` (or maybe a whitelist instead?)

**Comment by memeplex** [comment]
Why I'm being flooded by linting done on my conda environments:

![image](https://user-images.githubusercontent.com/2845433/120940824-85434e80-c6f5-11eb-8bbf-24cab8038bdb.png)

I have `openFilesOnly` and pickle.py is not open. Besides, the environment is not inside my working folder nor my repo. 

It's not an exceptional case, I get this kind of spurious diagnostic all the time for my environments.

**Comment by memeplex** [comment]
I'm getting spurious warnings from linting in my conda environments:

![image](https://user-images.githubusercontent.com/2845433/120940944-4792f580-c6f6-11eb-830f-3c721e9265bf.png)

I get `openFilesOnly` set and pickle.py is not open. Moreover the environment is not inside my workspace nor repo. This is not an isolated occurrence, it happens very often.

**Comment by memeplex** [comment]
This might be useful: the reports appear after I run unit tests from inside vscode. I use pytest.
